From ba10ecfea365085d4302ebd589f10639bad474e6 Mon Sep 17 00:00:00 2001
From: Pavankumar Nandeshwar <pnandesh@qti.qualcomm.com>
Date: Fri, 20 Jun 2025 15:01:24 +0530
Subject: [PATCH 18/21] UPSTREAM: DP: Remve dp_lock while fetching peer in rx
 path

Remve dp_lock while fetching peer_id in rx path as the
peer is protected by the rcu lock

Patch-dependency : None

Signed-off-by: Pavankumar Nandeshwar <pnandesh@qti.qualcomm.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c       | 10 +---------
 drivers/net/wireless/ath/ath12k/wifi7/dp_rx.c |  4 +---
 2 files changed, 2 insertions(+), 12 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index bca0e9d..04d7b87 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -206,18 +206,14 @@ static void ath12k_dp_rx_mld_addr_conv(struct ath12k_pdev_dp *dp_pdev,
 				       struct sk_buff *msdu,
 				       struct hal_rx_desc *rx_desc, u16 peer_id)
 {
-	struct ath12k_base *ab = dp_pdev->ar->ab;
 	struct ath12k_dp_link_peer *peer;
 	struct ieee80211_hdr *hdr = (void *)msdu->data;
 
-	spin_lock_bh(&ab->base_lock);
 	peer = ath12k_dp_rx_h_find_peer(dp_pdev->dp, rx_desc, peer_id);
 	if (!peer || !peer->mlo) {
-		spin_unlock_bh(&ab->base_lock);
 		return;
 	}
 	ether_addr_copy(hdr->addr2, peer->ml_addr);
-	spin_unlock_bh(&ab->base_lock);
 }
 
 void ath12k_dp_rx_h_undecap_raw(struct ath12k_pdev_dp *dp_pdev, struct sk_buff *msdu,
@@ -944,15 +940,12 @@ ath12k_dp_rx_h_find_peer(struct ath12k_dp *dp, struct hal_rx_desc *rx_desc, u16
 	struct ath12k_dp_link_peer *peer = NULL;
 	void *peer_mac;
 
-	lockdep_assert_held(&dp->dp_lock);
 
 	peer = ath12k_dp_link_peer_find_by_id(dp, peer_id);
-
 	if (peer)
 		return peer;
 
 	peer_mac = ath12k_hal_rxdesc_get_mpdu_start_addr2(dp->hal, rx_desc);
-
 	if (peer_mac)
 		peer = ath12k_dp_link_peer_find_by_addr(dp, peer_mac);
 
@@ -972,7 +965,7 @@ void ath12k_dp_rx_deliver_msdu(struct ath12k_pdev_dp *dp_pdev,
 	struct ath12k_dp_peer *peer;
 
 	rcu_read_lock();
-	spin_lock_bh(&dp->dp_lock);
+
 	peer = ath12k_dp_peer_find_by_peerid_index(dp, dp_pdev, peer_id);
 
 	pubsta = peer ? peer->sta : NULL;
@@ -996,7 +989,6 @@ void ath12k_dp_rx_deliver_msdu(struct ath12k_pdev_dp *dp_pdev,
 		}
 	}
 
-	spin_unlock_bh(&dp->dp_lock);
 	rcu_read_unlock();
 
 	ath12k_dbg(ab, ATH12K_DBG_DATA,
diff --git a/drivers/net/wireless/ath/ath12k/wifi7/dp_rx.c b/drivers/net/wireless/ath/ath12k/wifi7/dp_rx.c
index a56f1a8..e0fbe24 100644
--- a/drivers/net/wireless/ath/ath12k/wifi7/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/wifi7/dp_rx.c
@@ -893,7 +893,7 @@ static int ath12k_wifi7_dp_rx_h_mpdu(struct ath12k_pdev_dp *dp_pdev,
 	ath12k_wifi7_dp_rx_h_csum_offload(msdu, rx_msdu_info);
 
 	rcu_read_lock();
-	spin_lock_bh(&dp->dp_lock);
+
 	peer = ath12k_dp_peer_find_by_peerid_index(dp, dp_pdev, peer_id);
 	if (peer) {
 		/* restting 4addr da mcbc packets as in 4addr mcbc packets are
@@ -909,7 +909,6 @@ static int ath12k_wifi7_dp_rx_h_mpdu(struct ath12k_pdev_dp *dp_pdev,
 			ath12k_wifi7_dp_rx_check_fast_rx(dp->ab, msdu, rx_msdu_info,
 							 tlv_info, link_peer)) {
 			msdu->protocol = eth_type_trans(msdu, peer->dev);
-			spin_unlock_bh(&dp->dp_lock);
 			rcu_read_unlock();
 			netif_receive_skb(msdu);
 			return ret;
@@ -931,7 +930,6 @@ static int ath12k_wifi7_dp_rx_h_mpdu(struct ath12k_pdev_dp *dp_pdev,
 		enctype = HAL_ENCRYPT_TYPE_OPEN;
 	}
 
-	spin_unlock_bh(&dp->dp_lock);
 	rcu_read_unlock();
 
 	*fast_rx = false;
-- 
2.34.1

